File: src/main/java/arriba/fix/session/InMemorySessionResolver.java
package arriba.fix.session;

import java.util.Map;

public final class InMemorySessionResolver implements SessionResolver {

    private final Map<SessionId, Session> sessionIdToSession;

    public InMemorySessionResolver(final Map<SessionId, Session> sessionIdToSession) {
        this.sessionIdToSession = sessionIdToSession;
    }

    public Session resolve(final SessionId sessionId) {
        return this.sessionIdToSession.get(sessionId);
    }
}

File: src/main/scala/arriba/scala/fix/session/InMemorySessionResolver.scala
package arriba.scala.fix.session

import java.util.Map

final class InMemorySessionResolver(sessionIdToSession: Map[SessionId, Session]) extends SessionResolver {

  def resolve(sessionId: SessionId): Session = {
    val session: Session = this.sessionIdToSession.get(sessionId)
    if (session == null) {
      throw new Exception("Unknown session ID:  " + sessionId + ".")
    }
    session
  }
}

